app/models/manageiq/providers/openstack/base_metrics_capture.rb

```diff
@@ -92,30 +92,34 @@ def list_meters(filter)
   def meter_names(metric_capture_module)
     @meter_names ||= metric_capture_module::COUNTER_NAMES
   end
+  
+  cache_with_timeout(:counters_by_vm, 60.minutes) { Hash.new }
 
   def find_meter_counters(metric_capture_module, resource_filter, metadata_filter, log_header)
-    counters = list_resource_meters(resource_filter, log_header) + list_metadata_meters(metadata_filter, log_header)
-
-    # With Gnocchi, the network metrics are not associated with the instance's resource id
-    # but with the instance's network interface resource id. Here we fetch the counters
-    # for the network interface, so that the network metrics can be fetched.
-    if target.respond_to?(:network_ports)
-      target.network_ports.each do |port|
-        # fetch the list of resources and use the original_resource_id and type to find
-        # the network interface's resource
-        if port.ems_ref
-          original_resource_id = "#{target.ems_ref}-tap#{port.ems_ref[0..10]}"
-          resources = @perf_ems.list_resources("instance_network_interface").body
-          resources.each do |r|
-            if r["original_resource_id"].include?(original_resource_id)
-              resource_filter = {"field" => "resource_id", "value" => r["id"]}
-              counters = counters + list_resource_meters(resource_filter, log_header)
+    counters = self.class.counters_by_vm.dig(target.ems_id, target.ems_ref)
```

Comment:
This was `ems.id` but since on ivanchuk we don't have an `@ems` ivar I just used the `target.ems_id`